CMakeLists.txt

```diff
@@ -87,259 +147,83 @@ if(CMAKE_SYSTEM_NAME STREQUAL SunOS)
     OUTPUT_VARIABLE SUNOS_NATIVE_INSTRUCTION_SET
     )
   if(SUNOS_NATIVE_INSTRUCTION_SET MATCHES "amd64")
-    set(CLR_CMAKE_PLATFORM_UNIX_TARGET_AMD64 1)
+    set(CLR_CMAKE_PLATFORM_UNIX_AMD64 1)
     set(CMAKE_SYSTEM_PROCESSOR "amd64")
   else()
     clr_unknown_arch()
   endif()
   set(CLR_CMAKE_PLATFORM_SUNOS 1)
 endif(CMAKE_SYSTEM_NAME STREQUAL SunOS)
 
-if(CLR_CMAKE_PLATFORM_UNIX_TARGET_ARM)
+#--------------------------------------------
+# This repo builds two set of binaries
+# 1. binaries which execute on target arch machine
+#        - for such binaries host architecture & target architecture are same
+#        - eg. coreclr.dll
+# 2. binaries which execute on host machine but target another architecture
+#        - host architecture is different from target architecture
+#        - eg. crossgen.exe - runs on x64 machine and generates nis targeting arm64
+#        - for complete list of such binaries refer to file crosscomponents.cmake
+#-------------------------------------------------------------
+# Set HOST architecture variables
+if(CLR_CMAKE_PLATFORM_UNIX_ARM)
+  # host arch is arm
```

Comment:
I disagree. It might be intuitive for you but it would help any new person reading the code. It explains why there is a need for host arch & target arch and when they would be different.